internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -469,6 +465,17 @@ func resourceMsSqlVirtualMachine() *pluginsdk.Resource {
 			"tags": commonschema.Tags(),
 		},
 	}
+
+	if !features.FivePointOh() {
+		resource.Schema["auto_backup"].Elem.(*pluginsdk.Resource).Schema["encryption_enabled"] = &pluginsdk.Schema{
+			Type:       pluginsdk.TypeBool,
+			Optional:   true,
+			Default:    false,
+			Deprecated: "This argument is no longer used. Encryption is enabled when encryption_password is set; otherwise disabled.",
```

Comment:
I missed this previously, but `Default: false` causes an inaccurate diff if you remove `encryption_enabled` from your config after you've created the virtual machine with encryption enabled. (it will say `encryption_enabled: true -> false` even though the request sent to Azure will have encryption enabled)

With deprecation messages we tend to be very specific as well. 2 minor suggestions on the schema here:

```suggestion
		resource.Schema["auto_backup"].Elem.(*pluginsdk.Resource).Schema["encryption_enabled"] = &pluginsdk.Schema{
			Type:       pluginsdk.TypeBool,
			Optional:   true,
			Computed:   true,
			Deprecated: "`encryption_enabled` has been deprecated and will be removed in v5.0 of the AzureRM Provider. Encryption is enabled when `encryption_password` is set; otherwise disabled.",
```

Reply:
Applied the change. Good catch